intezer_sdk/endpoint_analysis.py

```diff
@@ -62,3 +89,136 @@ def _init_sub_analyses(self):
+        finally:
+            if self.status == consts.AnalysisStatusCode.IN_PROGRESS:
+                self._scan_api.close_scan_store(scan_summary={'reason': EndpointAnalysisEndReason.DONE.value})
+                self.status = consts.AnalysisStatusCode.CREATED
+        return self.analysis_id
+
+    def _create_scan(self):
+        with open(os.path.join(self._metadata_dir, 'scanner_info.json')) as f:
+            scanner_info = json.load(f)
+        scanner_info['scan_type'] = SCAN_TYPE_OFFLINE_ENDPOINT_SCAN
+        result = self._api.create_endpoint_scan(scanner_info)
+        self.scan_id = result['scan_id']
+        self.analysis_id = result['analysis_id']
+
+    def _initialize_endpoint_api(self):
+        if not self._scan_api:
+            self._scan_api = EndpointScanApi(self.scan_id, self._api)
+
+    def _send_host_info(self):
+        logger.info('Sending host info')
+        with open(os.path.join(self._metadata_dir, 'host_info.json')) as f:
+            host_info = json.load(f)
+        self._scan_api.send_host_info(host_info)
+
+    def _send_processes_info(self):
+        logger.info('Sending processes info')
+        with open(os.path.join(self._metadata_dir, 'processes_info.json')) as f:
+            processes_info = json.load(f)
+        self._scan_api.send_processes_info(processes_info)
+
+    def _send_scheduled_tasks_info(self):
+        if not os.path.exists(os.path.join(self._metadata_dir, 'scheduled_tasks_info.json')):
+            return
+        logger.info('Sending scheduled tasks info')
+        with open(os.path.join(self._metadata_dir, 'scheduled_tasks_info.json')) as f:
+            scheduled_tasks_info = json.load(f)
+        self._scan_api.send_scheduled_tasks_info(scheduled_tasks_info)
+
+    def _send_loaded_modules_info(self):
+        logger.info('Sending loaded modules info')
+        for loaded_module_info_file in glob.glob(os.path.join(self._metadata_dir, '*_loaded_modules_info.json')):
+            with open(loaded_module_info_file, 'r') as f:
+                loaded_modules_info = json.load(f)
+
+            pid = os.path.basename(loaded_module_info_file).split('_')[0]
+            self._scan_api.send_loaded_modules_info(pid, loaded_modules_info)
+
+    def _send_files_info_and_upload_required(self):
+        logger.info('Sending files info and uploading required files')
+        for files_info_file in glob.glob(os.path.join(self._metadata_dir, 'files_info_*.json')):
+            with open(files_info_file, 'r') as f:
+                files_info = json.load(f)
+
+            files_to_upload = self._scan_api.send_files_info(files_info)
+            with ThreadPoolExecutor() as executor:
+                for file_to_upload in files_to_upload:
+                    file_path = os.path.join(self._files_dir, file_to_upload + '.sample')
+                    if os.path.exists(file_path):
+                        executor.submit(self._scan_api.upload_collected_binary, file_path, 'file-system')
```

Comment:
you'll have to monitor the results, an upload might fail and the scan will continue as usually